dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletAPI.java

```diff
@@ -247,7 +247,25 @@ public Contentlet copyContentlet(Contentlet contentlet, Host host, User user, bo
 	 * @throws DotContentletStateException
 	 */
 	public Contentlet copyContentlet(Contentlet contentlet, Folder folder, User user, boolean appendCopyToFileName, boolean respectFrontendRoles) throws DotDataException, DotSecurityException, DotContentletStateException;
-	
+
+	/**
+	 * Copies a contentlet, including all its fields including binary files, image and file fields are pointers and the are preserved as the are
+	 * so if source contentlet points to image A and resulting new contentlet will point to same image A as well, also copies source permissions.
+	 * And moves the the new piece of content to the given folder. CopySuffix will be to append suffix to the file name.
+	 *
+	 * @param contentletToCopy
+	 * @param host
+	 * @param folder
+	 * @param user
+	 * @param copySuffix
+	 * @param respectFrontendRoles
+	 * @return Contentlet
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 * @throws DotContentletStateException
+	 */
+	Contentlet copyContentlet(Contentlet contentletToCopy, Host host, Folder folder, User user, final String copySuffix, boolean respectFrontendRoles) throws DotDataException, DotSecurityException, DotContentletStateException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.dotmarketing.portlets.contentlet.business.DotContentletStateException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 223 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)